iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -590,4 +643,135 @@ private static Schema schemaWithoutConstantsAndMeta(Schema readSchema, Map<Integ
       return TypeUtil.selectNot(readSchema, collect);
     }
   }
+
+  private static final class IcebergMergeRecordReader<T> extends AbstractIcebergRecordReader<T> {
```

Comment:
why don't we move this into a standalone class

Reply:
Done.